subsys/pm/device_runtime.c

```diff
@@ -24,6 +25,31 @@ LOG_MODULE_DECLARE(pm_device, CONFIG_PM_DEVICE_LOG_LEVEL);
 
 #define EVENT_MASK		(EVENT_STATE_ACTIVE | EVENT_STATE_SUSPENDED)
 
+static int invoke_suspend(const struct device *dev)
+{
+	pm_policy_device_power_lock_put(dev);
+	pm_device_domain_children_turn_off(dev);
```

Comment:
AFAICT this pattern is still adding a chunk of overhead to all PM calls.

Every suspend/resume cycle is iterating over every child node regardless of whether the device has any capability of acting as a power domain. This is particularly problematic for `PM_DEVICE_FLAG_ISR_SAFE`, which is running all of these operations in an ISR context.

Should this not be using `PM_DEVICE_FLAG_PD` in order to determine whether to run these functions in the first place?
This would also allow dropping the `pm: device: pm_device_children_action_run: validate child domain` commit entirely, as I can't see any way in which a device could be a child of a power domain without being on that power domain.